include/internal/param_names.h.in

```diff
@@ -0,0 +1,47 @@
+/*
+ * {- join("\n * ", @autowarntext) -}
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+{-
+use OpenSSL::paramnames qw(generate_internal_macros);
+-}
+
+#define MAX_PARAM_COUNT     100
```

Comment:
Couldn't this `NUM_PIDX`?

Reply:
... that way, the array boundary wouldn't be much of a worry, right? (we could still do with an assertion to make sure we don't do stupid things, but yeah)

Reply:
Some parameters can be repeated and concatenated, so no finite limit is perfect.

Reply:
Gah, you're right, I forgot about that.